src/components/BusinessDonationHistory/ViewDonationHistory/ViewDonationHistory.jsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable react/prop-types */
+
+import { useEffect, useState } from 'react';
+import { Table, Tbody, Tr, Td } from '@chakra-ui/react';
+import { useBackend } from '../../../contexts/BackendContext';
+
+const ViewDonationHistory = ({ id }) => {
+  const [data, setData] = useState([]);
+  const { backend } = useBackend();
+
+  const fetchDonation = async () => {
+    try {
+      const response = await backend.get(`/donation/${id}`);
+      setData(response.data);
+    } catch (error) {
+      console.error('Error:', error);
+    }
+  };
+
+  useEffect(() => {
+    if (id) {
+      fetchDonation();
+    }
+  }, [id]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'fetchDonation'. Either include it or remove the dependency array.